### DISCUSSION TITLE
An option to add fields into relevant groups without dragging them.

### DISCUSSION THREAD

**Discussion by itszeeshan** [directus/directus#23168] [Feature Requests]
### Summary

Hi,

Whenever we create groups in UI data model section, the groups are empty by default. Then we had to first create fields and then drag them into relevant groups. Or we can first create all fields and then make groups and put them into their relevant groups. Now this 1 by 1 is a hassle if the fields in the collection are too much. So to decrease this hassle and make it more easy for the users we can(I hope we can) implement this:

#### _Solution for Case 1 (fields are already created and now we are creating groups):_
* While creating group field, in its advance search we enlist names of all the fields that are currently present in the collection.
* Those enlisted names are checkboxes which user can click to add fields into the group.
* Just like we have in permissions page, when we are adding custom permission for fields.

#### _Solution for Case 2 (groups are already present use is now creating fields):_
* User can see the groups in the advance details and can click in any one of them to be in that group from start.

####  _Solution for Case 3 (groups are already and so are the fields):_
* User can use this menu, sees another option along with these current options, there named as "drag to group", hover over this button that opens a list of groups we created already. Click in any one of them to put the field into that group.
<img width="989" alt="Screenshot 2024-07-25 at 6 51 23 AM" src="https://github.com/user-attachments/assets/3b52c232-b668-4267-ad6e-cadb329f2ae5">

### Motivation

* To save user time
* So user wont get irritated if the fields are too much
* To basically remove the hassle :)

### Detailed Design

Design present in summary section above with detailed cases that includes user interactions.

#### _Advanced Search with Checkboxes for Field Selection (Case 1):_
* When creating a group, users should see a list of all existing fields with checkboxes, allowing them to add fields to the group easily.
#### _Group Selection in Field Creation (Case 2):_
* While creating a new field, users should see the existing groups in the advanced details and be able to assign the field to a group immediately.
#### _Drag to Group Feature (Case 3):_
* Users should have an option to drag fields to groups through a menu that appears on hovering, listing all existing groups for easy assignment.
### Won't Have:
* Any feature not explicitly mentioned in the solutions provided, such as auto-grouping based on field names or automatic suggestions for grouping.

### Drawbacks

* If the number of fields is very large, listing all of them with checkboxes might make the interface cluttered and slow (at case 1)
* Loading a large number of fields in the advanced search could impact the performance and responsiveness of the UI.

#### Multi-Select Dropdown at Case 1
* Instead of a list with checkboxes, use a multi-select dropdown menu. Users can type to filter the field names and select multiple fields at once.

### Adoption Strategy

* Analyze if the new grouping features can coexist with the current functionality without breaking existing workflows. Determine if any changes to the Directus API are necessary and ensure they do not disrupt existing integrations.

* Consider a mode that supports both old and new methods temporarily to allow gradual migration.

* Release Notes:
 Clearly document the changes, including any breaking changes, in the release notes.
* Migration Guides:
 Provide detailed migration guides for developers to transition their projects smoothly.

**Comment by JayShoe** [comment]
I have a table with 800 fields, and organizing them in the current UI is painfully slow. A **table view** with **multi-select** and **bulk actions** would make field management much faster and easier for large schemas—a simple but impactful change.

**Reply by rijkvanzanten** [reply]
It'd have to be a nested tree-table though as things are grouped. Impactful for sure, but not as simple to just whip up 😅

**Reply by JayShoe** [reply]
Not necessarily. I'm envisioning something like a table view, similar to the table collection view. Each column would represent an attribute of the field, such as:

- **Field Label**
- **Field Name**
- **Field Type** (e.g., String, Checkbox, etc.)
- **Field Group**

While this view wouldn’t support drag-and-drop sorting by default, it would allow you to sort columns, effectively grouping fields by attributes like "Field Group." From there, you could select fields to edit their attributes or make other changes aside from sorting. When you're ready to rearrange the fields, you could switch back to the tile view to organize in the standard way.

**Reply by rijkvanzanten** [reply]
Gotcha gotcha, so making it quite a bit more "Database Manager" style. F.e. tableplus:

![CleanShot 2024-11-26 at 18 36 46@2x](https://github.com/user-attachments/assets/5b3627b0-4cfc-4712-8770-075ce9d50ec7)

**Reply by itszeeshan** [reply]
Exactly @rijkvanzanten , And I totally agree with @JayShoe; I really like this idea of having a table view, and we would need pagination ofcourse and handle a lot of things more carefully especially for 800 fields types of cases 😅. However, I'm not entirely sure about one thing, how should groups appear in the table? Should they be their own rows or just a column? and If groups are nested, how should we represent that in the table? And if we are providing user with a table view it would be better to have bulk actions like (e.g., assign fields to a group, toggle visibility, delete fields in bulk) and more.

**Reply by rijkvanzanten** [reply]
> However, I'm not entirely sure about one thing, how should groups appear in the table? Should they be their own rows or just a column? and If groups are nested, how should we represent that in the table?

This is exactly what I meant by the "It'd have to be a tree-table" 😄 Feels like in order for groups to be represented well, you'd kinda want to have them nested in the table as well right? 🤔 

Something like:

```
name             type        default
--------------------------------------
content_group    alias       --
 ├ title         varchar     Hello World
 ├ body          text        --
another_field    etc         
```

**Reply by itszeeshan** [reply]
Yeah I guess you're right 😄

**Comment by JayShoe** [comment]
@itszeeshan @rijkvanzanten 

I appreciate keeping things as straightforward as possible. While grouped tables may seem appealing at first glance, they introduce added complexity for core actions like `sort` or `select all`, which can become challenging in a grouped context. Instead, I suggest using other views to achieve the desired groupings. I fully support having multiple views, similar to how content is managed. Plus, as mentioned, you can use `sort` to group items effectively anyway. I think @rijkvanzanten is correct with his "database manager" style for sure. With the ability to add or remove columns, sort, and search (if I'm being greedy). The search allows you to filter, select all, and apply groupings easily too, FWIW.

In the **Flow Manager** extension, they had a tree based view that I found to be a little difficult to work with.  I requested a table view—and to my delight, it was implemented! 😀 I think it works wonderfully as a simple and effective way to view content. In this screenshot you'll see I'm not grouping with Categories, I'm sorting by Name. 

![image](https://github.com/user-attachments/assets/87d753b1-450d-450c-8998-4727e7981304)

Honestly once you go to a tree view you basically have what you have now. It's not easy to move things around this UI either. 

![image](https://github.com/user-attachments/assets/b4856af0-9464-496c-891a-22b84281a83e)

**Reply by rijkvanzanten** [reply]
Yeah agreed. And that doesn't even get into pagination yet, which quickly becomes nightmarishly complicated once you have to start paginating the children of each row individually 🥴
